tests/slow/src/test/scala/tests/feature/Scala3CodeActionLspSuite.scala

```diff
@@ -259,6 +259,127 @@ class Scala3CodeActionLspSuite
        |""".stripMargin
   )
 
+  check(
+    "extract-if-cond",
+    """|object Main:
+       |  def main(i : Int) =
+       |    if(2 > <<3>>) 5 else 4
+       |
+       |""".stripMargin,
+    s"""|${ExtractValueCodeAction.title}
+        |""".stripMargin,
+    """|object Main:
+       |  def main(i : Int) =
+       |    val newValue = 2 > 3
+       |    if(newValue) 5 else 4
+       |
+       |""".stripMargin
+  )
+
+  check(
+    "extract-if-res",
+    """|object Main:
+       |  def main(i : Int) =
+       |  if(2 > 3) 5 <<+>> 1 else 4
+       |
+       |""".stripMargin,
+    s"""|${ExtractValueCodeAction.title}
+        |""".stripMargin,
+    """|object Main:
+       |  def main(i : Int) =
+       |    val newValue = 5 + 1
+       |    if(2 > 3) newValue else 4
+       |
+       |""".stripMargin
+  )
+
+  check(
+    "extract-tuple",
+    """|object Main:
+       |  def main(i : Int) =
+       |    val a = (1,<<2>>,3)
+       |
+       |""".stripMargin,
+    s"""|${ExtractValueCodeAction.title}
+        |""".stripMargin,
+    """|object Main:
+       |  def main(i : Int) =
+       |    val newValue = 2
+       |    val a = (1,newValue,3)
+       |
+       |""".stripMargin
+  )
+
+  check(
+    "extract-match",
+    """|object Main:
+       |  def main(i : Int) =
+       |    1 + <<2>> + 3 match {
+       |      case _ => 6
+       |    }
+       |
+       |""".stripMargin,
```

Comment:
I wonder how it will behave for `1 + <<(1 + 2)>> + 3` (I haven't look at the code yet)

Reply:
I wonder, wouldn't extracting first valid thing which encloses current range be a better behavior? Moreover, it would be awesome to have information what is being extracted in code action message - like `extract 1 + 2 + 3 to value`

Reply:
> I wonder, wouldn't extracting first valid thing which encloses current range be a better behavior? 

Maybe only if a range is specified? So if no range is specified then the last enclosing thing, but when range is specified then just the last thing that fits? I would do that in a separate PR though. 

Adding a title might be a good idea, but let's truncate it if it's too long

Reply:
> Adding a title might be a good idea, but let's truncate it if it's too long

Yeah, truncating to e.g. 10 character and then showing ellipsis (`...`) should be enough to show what will be extracted